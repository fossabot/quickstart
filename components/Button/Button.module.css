.Button {
  display: inline-block;
  font-weight: 700;
  font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  line-height: 1;
  border: 0;
  border-radius: 3em;
  cursor: pointer;
}

.Primary {
  color: white;
  background-color: #1ea7fd;
}

.Secondary {
  color: #333;
  background-color: transparent;
  box-shadow: rgba(0, 0, 0, 0.15) 0 0 0 1px inset;
}

.Small {
  padding: 10px 16px;
  font-size: 12px;
}

.Medium {
  padding: 11px 20px;
  font-size: 14px;
}

.Large {
  padding: 12px 24px;
  font-size: 16px;
}
