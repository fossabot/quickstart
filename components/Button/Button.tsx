import React from 'react'

import Styles from './Button.module.css'

export interface ButtonProps {
  /**
   * Is this the principal call to action on the page?
   */
  primary?: boolean
  /**
   * What background color to use
   */
  backgroundColor?: string
  /**
   * How large should the button be?
   */
  size?: 'small' | 'medium' | 'large'
  /**
   * Button contents
   */
  label: string
  /**
   * Optional click handler
   */
  onClick?: () => void
}

/**
 * Primary UI component for user interaction
 */
export const Button: React.FC<ButtonProps> = ({
  primary = false,
  size = 'Medium',
  backgroundColor,
  label,
  ...props
}) => {
  const modeClass = primary ? Styles.Primary : Styles.Secondary
  const sizeClass =
    size === 'small'
      ? Styles.Small
      : size === 'large'
      ? Styles.Large
      : Styles.Medium

  return (
    <button
      type="button"
      className={[Styles.Button, sizeClass, modeClass].join(' ')}
      style={{ backgroundColor }}
      {...props}
    >
      {label}
    </button>
  )
}
