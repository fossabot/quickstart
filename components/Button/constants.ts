export const STYLES_BASE =
  'inline-block font-bold leading-4 border-solid border-2 rounded'
export const STYLES_PRIMARY = 'text-white bg-blue-600 border-blue-700'
export const STYLES_SECONDARY = 'bg-transparent border-gray-600'

export const STYLES_MEDIUM = 'text-base px-4 py-4'
export const STYLES_SMALL = 'text-sm px-3 py-3'
export const STYLES_LARGE = 'text-lg px-6 py-6'
