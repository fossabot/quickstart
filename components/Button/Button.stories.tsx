// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Meta, Story } from '@storybook/react/types-6-0'
import React from 'react'

import { Button, ButtonProps } from './Button'

const config: Meta = {
  title: 'Components/Button',
  component: Button,
  argTypes: {
    backgroundColor: { control: 'color' },
  },
}

export default config

const Template: Story<ButtonProps> = (args) => <Button {...args} />

export const Default = Template.bind({})
Default.args = {
  label: 'Button',
}
Default.parameters = {
  jest: ['Button.test.tsx'],
  design: {
    type: 'figma',
    url: 'https://www.figma.com/file/LKQ4FJ4bTnCSjedbRpk931/Sample-File',
  },
}

export const Primary = Template.bind({})
Primary.args = {
  primary: true,
  label: 'Button',
}

export const Secondary = Template.bind({})
Secondary.args = {
  label: 'Button',
}

export const Large = Template.bind({})
Large.args = {
  size: 'large',
  label: 'Button',
}

export const Small = Template.bind({})
Small.args = {
  size: 'small',
  label: 'Button',
}
