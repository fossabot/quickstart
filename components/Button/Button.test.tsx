import React from 'react'

import { Button, ButtonProps } from './Button'

describe('Button', () => {
  const props: ButtonProps = {
    label: 'Test Button',
  }
  it('Should render without an error', () =>
    expect(() => <Button {...props} />).not.toThrow())

  it('Should match the snapshot', () =>
    expect(<Button {...props} />).toMatchSnapshot())
})
