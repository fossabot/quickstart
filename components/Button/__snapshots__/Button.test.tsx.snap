// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match the snapshot 1`] = `
<Button
  label="Test Button"
/>
`;
